src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1451,9 +1446,9 @@ abstract class RefChecks extends Transform {
             )
       }
 
-      sym.isSourceMethod &&
+      sym.name == nme.apply &&
+        !(sym hasFlag STABLE) && // ???
```

Comment:
The use of `STABLE` to identify compiler-created methods dates back to https://github.com/scala/scala/commit/9acfa7693d146fc2ea6fdad61c74907354910ea8#diff-d5789e5ae5061197d782d08324b260dbR97.

Through refactorings and comment ping pong, Martin commented that it was to "exclude all accessors" from the designation of `isSourceMethod`. 

In this instance, I think the subsequent check of `isCase` makes the check for `!STABLE` redundant.

```
scala> case class C(apply: Int) { }; object C { private val apply = 42 }
defined class C
defined object C

scala> :power
Power mode enabled. :phase is at typer.
import scala.tools.nsc._, intp.global._, definitions._
Try :help or completions for vals._ and power._

scala> typeOf[C].member("apply").isSourceMethod
<console>:39: error: type mismatch;
 found   : String("apply")
 required: $r.intp.global.Name
Note that implicit conversions are not applicable because they are ambiguous:
 both method stringToTermName in trait Names of type (s: String)$r.intp.global.TermName
 and method stringToTypeName in trait Names of type (s: String)$r.intp.global.TypeName
 are possible conversion functions from String("apply") to $r.intp.global.Name
       typeOf[C].member("apply").isSourceMethod
                        ^

scala> typeOf[C].member(TermName("apply")).isSourceMethod
res1: Boolean = false

scala> typeOf[C].member(TermName("apply")).isCase
res2: Boolean = false

scala> typeOf[C.type].member(TermName("apply")).alternatives
res6: List[$r.intp.global.Symbol] = List(method apply, value apply)

scala> typeOf[C.type].member(TermName("apply")).alternatives.map(_.isCase)
res7: List[Boolean] = List(true, false)
```